### PR TITLE
[IMP] stock,*: Adapts stock related demo data to be more convenient with default config

### PR DESCRIPTION
**Pull request by svs-odoo**
See commits' messages for more informations.

task-1970460

### CODE REVIEW

File: addons/stock/data/stock_demo.xml [hunk 1]
```diff
@@ -157,7 +170,7 @@
             <field name="company_id" eval="False"/>
         </record>
 
-        <record id="res_company_1" model="res.company">
+        <record id="res_company_1" model="res.company" context="{'default_code': 'CHIC1'}">
```
Comment (amoyaux) [inline addons/stock/data/stock_demo.xml]: with this https://github.com/odoo/odoo/pull/47201 and another small fix you could probably get rid of this hack and move it in the write bellow

File: addons/stock/tests/test_robustness.py [hunk 2]
```diff
@@ -71,19 +71,23 @@ def test_location_usage(self):
         consistent with the `reserved_quantity` on the quants.
         """
         # change stock usage
-        self.stock_location.scrap_location = True
+        test_stock_location = self.env['stock.location'].create({
+            'name': "Test Location",
+            'location_id': self.stock_location.id,
+        })
+        test_stock_location.scrap_location = True
```
Comment (Whenrow) [inline addons/stock/tests/test_robustness.py]: put this directly in create()

File: addons/stock/data/stock_demo.xml [hunk 3]
```diff
@@ -185,11 +185,17 @@
             <field name="name">My Company (San Francisco)</field>
         </record>
 
+        <!-- Rename auto-created Chicago Warehouse -->
+        <function model="stock.warehouse" name="write">
+            <value model="stock.warehouse" search="[('id', '=', obj().env.ref('stock.res_company_1').id)]"/>
```
Comment (Whenrow) [inline addons/stock/data/stock_demo.xml]: isn't on company_id that you need to make the search  ?

File: addons/stock/data/stock_demo.xml (hunk 1 above)
Comment (Whenrow) [reply addons/stock/data/stock_demo.xml]: This comment is still accurate ?

File: addons/stock/tests/test_robustness.py (hunk 2 above)
Comment (svs-odoo) [reply addons/stock/tests/test_robustness.py]: That was already merged in the bugfix task :fearful:

**Comment by sle-odoo** [comment]
robodoo r+ rebase-ff

**Pull request closed without merging**